builder/relinker/relinker.py

```
@@ -0,0 +1,463 @@
+        def is_iram_desc(l):
+            # Recognize both original ldgen patterns and relinker-generated patterns
+            if '*(.iram1 .iram1.*)' in l:
+                return True
+            if ') .iram1 EXCLUDE_FILE(*' in l and ') .iram1.*)' in l:
+                return True
+            # Recognize relinker-generated IRAM exclude patterns
+            if '*(EXCLUDE_FILE(' in l and ') .iram1.*)' in l and ') .iram1)' in l:
+                return True
+            return False
+        
+        def is_relinker_iram_include(l):
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)